### ISSUE TITLE
StopIteration: 'NoneType' object has no attribute 'mode' 

### ISSUE THREAD

**Issue by anujnadig168** [keras-team/keras#10799]
Hello, 

I am running a simple Machine Learning algorithm to classify images into cat or dog images. This code was working perfectly until one day I got the error as mentioned in the title. Can anyone help me out with this? I have not found any solution to this. 

from keras.models import Sequential
from keras.layers import Dense
from keras.layers import Convolution2D
from keras.layers import MaxPooling2D
from keras.layers import Flatten

classifier = Sequential()

classifier.add(Convolution2D(32, 3, 3, input_shape = (64, 64, 3), activation = "relu"))

classifier.add(MaxPooling2D(pool_size = (2,2)))

classifier.add(Convolution2D(32,3,3, activation = "relu"))

classifier.add(MaxPooling2D(pool_size = (2,2)))

classifier.add(Flatten())

classifier.add(Dense(output_dim = 128, activation = "relu"))
classifier.add(Dense(output_dim = 1, activation = "sigmoid"))

classifier.compile(optimizer = 'adam', metrics = ['accuracy'], loss = 'binary_crossentropy')

from keras.preprocessing.image import ImageDataGenerator
train_datagen = ImageDataGenerator(rescale = 1./255,
                                   shear_range = 0.2,
                                   zoom_range = 0.2,
                                   horizontal_flip = True)

test_datagen = ImageDataGenerator(rescale = 1./255)

training_set = train_datagen.flow_from_directory('/Users/Anuj/Desktop/Machine Learning A-Z Template Folder/Part 8 - Deep Learning/Section 40 - Convolutional Neural Networks (CNN)/dataset/training_set',
                                                 target_size = (64,64),
                                                 batch_size = 32,
                                                 class_mode = "binary")

test_set = test_datagen.flow_from_directory('/Users/Anuj/Desktop/Machine Learning A-Z Template Folder/Part 8 - Deep Learning/Section 40 - Convolutional Neural Networks (CNN)/dataset/test_set',
                                                 target_size = (64,64),
                                                 batch_size = 32,
                                                 class_mode = "binary")

classifier.fit_generator(training_set,
                         samples_per_epoch = 8000,
                         nb_epoch = 25,
                         validation_data = test_set,
                         nb_val_samples = 2000)

import numpy as np
from keras.preprocessing import image
test_image = image.load_img('/Users/Anuj/Desktop/Machine Learning A-Z Template Folder/Part 8 - Deep Learning/Section 40 - Convolutional Neural Networks (CNN)/dataset/test_set/cats/cat.4020.jpg', target_size = (64, 64))
test_image = image.img_to_array(test_image)
test_image = np.expand_dims(test_image, axis = 0)
result = classifier.predict(test_image)
training_set.class_indices
if result[0][0] == 1:
    prediction = 'dog'
else:
    prediction = 'cat'

The errors I am getting are: 

  File "C:\Users\Anuj\Anaconda3\lib\site-packages\spyder\utils\site\sitecustomize.py", line 705, in runfile
    execfile(filename, namespace)

  File "C:\Users\Anuj\Anaconda3\lib\site-packages\spyder\utils\site\sitecustomize.py", line 102, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/Anuj/Desktop/Machine Learning A-Z Template Folder/Part 8 - Deep Learning/Section 40 - Convolutional Neural Networks (CNN)/CNN_own.py", line 54, in <module>
    nb_val_samples = 2000)

  File "C:\Users\Anuj\Anaconda3\lib\site-packages\keras\legacy\interfaces.py", line 91, in wrapper
    return func(*args, **kwargs)

  File "C:\Users\Anuj\Anaconda3\lib\site-packages\keras\engine\training.py", line 1415, in fit_generator
    initial_epoch=initial_epoch)

  File "C:\Users\Anuj\Anaconda3\lib\site-packages\keras\engine\training_generator.py", line 177, in fit_generator
    generator_output = next(output_generator)

  File "C:\Users\Anuj\Anaconda3\lib\site-packages\keras\utils\data_utils.py", line 584, in get
    six.raise_from(StopIteration(e), e)

  File "<string>", line 3, in raise_from

StopIteration: 'NoneType' object has no attribute 'mode'
